examples/battleship/config.rs

```diff
@@ -31,11 +31,11 @@ impl Config {
 
     pub fn cell_color(cell: &Cell) -> Color {
         match cell {
-            Cell::Carrier => Self::CARRIER_COLOR.into(),
-            Cell::Battleship => Self::BATTLESHIP_COLOR.into(),
-            Cell::Cruiser => Self::CRUISER_COLOR.into(),
-            Cell::Submarine => Self::SUBMARINE_COLOR.into(),
-            Cell::Destroyer => Self::DESTROYER_COLOR.into(),
+            Cell::Ship(ShipKind::Carrier) => Self::CARRIER_COLOR.into(),
+            Cell::Ship(ShipKind::Battleship) => Self::BATTLESHIP_COLOR.into(),
+            Cell::Ship(ShipKind::Cruiser) => Self::CRUISER_COLOR.into(),
+            Cell::Ship(ShipKind::Submarine) => Self::SUBMARINE_COLOR.into(),
+            Cell::Ship(ShipKind::Destroyer) => Self::DESTROYER_COLOR.into(),
```

Comment:
Not really necessary but sometimes it is more readable to nest matches in that case:  

```rust
Cell::Ship(kind) => match kind {
            ShipKind::Carrier => Self::CARRIER_COLOR.into(),
            ShipKind::Battleship => Self::BATTLESHIP_COLOR.into(),
            ShipKind::Cruiser => Self::CRUISER_COLOR.into(),
            ShipKind::Submarine => Self::SUBMARINE_COLOR.into(),
            ShipKind::Destroyer => Self::DESTROYER_COLOR.into()
}